src/pages/ProfilePage.js

```diff
@@ -1,16 +1,128 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { Redirect } from "react-router-dom";
 
 import Container from "@material-ui/core/Container";
+import Divider from "@material-ui/core/Divider";
+import Grid from "@material-ui/core/Grid";
+import { makeStyles } from "@material-ui/core/styles";
 import Paper from "@material-ui/core/Paper";
 import Typography from "@material-ui/core/Typography";
+import EqualizerIcon from "@material-ui/icons/Equalizer";
 
-import User from "../components/User";
+import ProfileName from "../components/ProfileName";
+import UserStatistics from "../components/UserStatistics";
+import ProfileGamesTable from "../components/ProfileGamesTable";
+import useFirebaseRef from "../hooks/useFirebaseRef";
+import firebase from "../firebase";
+import { computeState } from "../util";
+import LoadingPage from "./LoadingPage";
+
+const useStyles = makeStyles((theme) => ({
+  divider: {
+    [theme.breakpoints.down("sm")]: {
+      display: "none",
+    },
+  },
+}));
 
 function ProfilePage({ match }) {
+  const userId = match.params.id;
+
+  const [games, loadingGames] = useFirebaseRef(`/userGames/${userId}`);
+  const [gamesData, setGamesData] = useState(null);
+  const [redirect, setRedirect] = useState(null);
+
+  const classes = useStyles();
+
+  const handleClickGame = (gameId) => {
+    setRedirect(`/room/${gameId}`);
+  };
+
+  useEffect(() => {
```

Comment:
This seems a little complicated. `mergeGameData` could be moved outside of the `useEffect`, right? It could be a top-level function instead since it doesn't depend on anything in the scope.